platforms/core-configuration/model-core/src/integTest/groovy/org/gradle/api/provider/CollectionPropertyIntegrationTest.groovy

```diff
@@ -233,8 +229,9 @@ task thing {
         "providers.provider { [ 'a', 'b', 'c' ] }" | _
     }
 
+    @Requires(value = IntegTestPreconditions.NotConfigCached, reason = "https://github.com/gradle/gradle/issues/27528")
```

Comment:
We fail to coherce `GString` to `String` in composite lazy properties (but not on non-composite ones).